include/micm/configure/camp_config.hpp

```diff
@@ -0,0 +1,786 @@
+      case ConfigParseStatus::Success: return "Success";
+      case ConfigParseStatus::None: return "None";
+      case ConfigParseStatus::InvalidKey: return "InvalidKey";
+      case ConfigParseStatus::UnknownKey: return "UnknownKey";
+      case ConfigParseStatus::InvalidCAMPFilePath: return "InvalidCAMPFilePath";
+      case ConfigParseStatus::NoConfigFilesFound: return "NoConfigFilesFound";
+      case ConfigParseStatus::CAMPFilesSectionNotFound: return "CAMPFilesSectionNotFound";
+      case ConfigParseStatus::CAMPDataSectionNotFound: return "CAMPDataSectionNotFound";
+      case ConfigParseStatus::InvalidSpecies: return "InvalidSpecies";
+      case ConfigParseStatus::InvalidMechanism: return "InvalidMechanism";
+      case ConfigParseStatus::ObjectTypeNotFound: return "ObjectTypeNotFound";
+      case ConfigParseStatus::RequiredKeyNotFound: return "RequiredKeyNotFound";
+      default: return "Unknown";
+    }
+  }
+
+  // Solver parameters
+  struct SolverParameters
+  {
+    System system_;
+    std::vector<Process> processes_;
+
+    SolverParameters(const System& system, std::vector<Process>&& processes)
+        : system_(system),
+          processes_(std::move(processes))
+    {
+    }
+
+    SolverParameters(System&& system, std::vector<Process>&& processes)
+        : system_(std::move(system)),
+          processes_(std::move(processes))
+    {
+    }
+  };
+
+  class JsonReaderPolicy
+  {
+    using json = nlohmann::json;
+
+    public:
+      std::vector<Species> species_arr_;
+
+      std::vector<UserDefinedRateConstant> user_defined_rate_arr_;
+      std::vector<ArrheniusRateConstant> arrhenius_rate_arr_;
+      std::vector<TroeRateConstant> troe_rate_arr_;
+      std::vector<BranchedRateConstant> branched_rate_arr_;
+      std::vector<SurfaceRateConstant> surface_rate_arr_;
+      std::vector<TernaryChemicalActivationRateConstant> ternary_rate_arr_;
+      std::vector<TunnelingRateConstant> tunneling_rate_arr_;
+
+      // Specific for solver parameters
+      Phase gas_phase_;
+      std::unordered_map<std::string, Phase> phases_;
+      std::vector<Process> processes_;
+
+      // Constants
+      // Configure files
+      static const inline std::string CAMP_CONFIG = "config.json";
+      static const inline std::string SPECIES_CONFIG = "species.json";
+      static const inline std::string MECHANISM_CONFIG = "mechanism.json";
+      static const inline std::string REACTIONS_CONFIG = "reactions.json";
+      static const inline std::string TOLERANCE_CONFIG = "tolerance.json";
+
+      // Common JSON
+      static const inline std::string CAMP_DATA = "camp-data";
+      static const inline std::string CAMP_FILES = "camp-files";
+      static const inline std::string TYPE = "type";
+
+      // Functions
+
+      /// @brief Parse configures
+      /// @param config_dir Path to a the configuration directory
+      /// @return True for successful parsing
+      ConfigParseStatus Parse(const std::filesystem::path& config_dir)
```

Comment:
In accordance with our [CCPP Standard Name](https://github.com/ESCOMP/CCPPStandardNames/pull/40#event-10757475398), I think we want to move to a single file rather than a directory. Then, we'd require that this file is essentially config.json. Pointing to a file will remove the need for hard coded filenames. Is that a change we can make now?

Reply:
Changed method Parse to take a single file path.  The config_dir is now the config_file.parent_path().